src/libraries/SignatureUtils.sol

```diff
@@ -16,16 +16,27 @@ library SignatureUtils {
 
     error InvalidSigner();
 
-    function verifySig(
+    function isValidSig(
         bytes32 _messageHash,
         address _signer,
         bytes32 _r,
         bytes32 _s,
         uint8 _v
-    ) public pure {
+    ) internal pure returns (bool) {
         bytes32 signedHash = _messageHash.toEthSignedMessageHash();
+        if (_signer != ECDSA.recover(signedHash, _v, _r, _s)) return false;
```

Comment:
can `_signer` be a contract supporting ERC-1271?
We can make the rest signers as multi-sigs, which makes the safe recovery as nested multi-sig structure